requirements.txt

```diff
@@ -11,3 +11,4 @@ certifi>=2023.11.17
 cython>=3.0.8
 cymem>=2.0.8
 orjson>=3.9.10
+typing-extensions>=4.7.0
```

Comment:
Even if it's already there as a transitive dependency of the WebHost (for Flask-Limiter), I'm not a fan. `TypeGuard` is available in `typing` since python 3.10, so I would rather have this marked as required for python less than 3.10. 

I think something like that would work:
```
typing-extensions>=4.7.0; python_version < '3.10'
```

Since typing-extensions won't be an explicit dependency for 3.10+, since it could be removed from transitive dependencies, `TypeGuard` could be imported with something like
```python
try:
  from typing import TypeGuard
except ImportError:
  from typing_extensions import TypeGuard
```

Reply:
That seems like adding complication for no benefit.

It's not like the usefulness of typing_extensions will go away at 3.10
There are also new features in 3.11 that would be useful (`Required` `NotRequired` `Self`)
and 3.12 (`@override`) and 3.13 (`ReadOnly`)
and there will be usefulness in 3.14, 3.15, 3.16...

No matter what version of Python we're on, it will be useful.

Reply:
Well, then I guess there is no point fighting against this dependency